packages/compass-schema-validation/src/modules/sample-documents.js

```diff
@@ -154,45 +169,51 @@ export const fetchSampleDocuments = (validator, error) => {
     }
 
     try {
-      const count = await dataService.count(namespace, query);
-
-      const docsOptions = {
-        namespace,
-        dataService,
-        count,
-      };
-
-      const matching = await getSampleDocuments({
-        ...docsOptions,
-        pipeline: [{ $match: query }, { $limit: 1 }],
-      });
-      const notmatching = await getSampleDocuments({
-        ...docsOptions,
-        pipeline: [{ $match: { $nor: [query] } }, { $limit: 1 }],
-      });
-      dispatch(
-        sampleDocumentsFetched({
-          matching: matching[0],
-          notmatching: notmatching[0],
+      const valid = (
+        await getSampleDocuments({
+          namespace,
+          dataService,
+          pipeline: [{ $match: query }, { $limit: 1 }],
         })
-      );
+      )[0];
+
+      dispatch(fetchedValidDocument(valid));
     } catch (e) {
-      setZeroDocuments(dispatch);
-      setSyntaxError(dispatch, e);
+      dispatch(fetchingValidDocumentFailed());
+      dispatch(syntaxErrorOccurred(e));
```

Comment:
It's good that you renamed this one to look less like "just a setter" and more like an event, which is definitely more in line with redux style guides, but even better would be to not emit multiple events here. You already emit one indicating that error occured, another slice reducer [can just handle it](https://redux.js.org/style-guide/#allow-many-reducers-to-respond-to-the-same-action)

Reply:
Nice, really liked the linked article. Thanks for pointing that out and also the reference. I will make the changes accordingly 😄

Reply:
After re-reading the flow - realised that we shouldn't even be dispatching `syntaxError` as per its implementation. It pops up the error in validation editor which probably is not the right place to show the errors related to fetching of documents. Additionally it gives wrong hint that the validation was modified which is not the case.

As far as the error originating from the check for validator goes - that is something we already show in the validation editor. For the errors particular to the fetch of documents - I can have a chat with our design team and get some input on that.

Reply:
Addressed by removing the dispatch of syntaxError because it was leading to a possibly incorrect state of validation editor.